### DISCUSSION TITLE
Inquiry Regarding Inconsistent Error in Simu5G 1.2.2 Scenarios with MEC

### DISCUSSION THREAD

**Discussion by rimsayegh** [Unipisa/Simu5G#198] [Q&A] [answered]
I am currently working with Simu5G version 1.2.2 and have encountered a recurring issue that I hope you can assist me with.

The problem arises in scenarios involving MEC, specifically with multiMecHost, SingleMecHost, and when adding more than two MECs. An error consistently appears. However, this error's occurrence is erratic; it happens intermittently, even when the same code and parameters are used. This inconsistency is observed both on Windows and Ubuntu platforms.

The error is: Invalid signal signalID=419 -- in module (ServiceRegistry) MultiMecHost.mecHost2.mecPlatform.serviceRegistry (id=679), at t=1.047366760966s, event#12307

Could you please provide some insight into the potential cause of this issue? Understanding the root of this problem is crucial for my ongoing work. 

<img width="960" alt="error with Service Registry" src="https://github.com/Unipisa/Simu5G/assets/87598361/26eba559-a600-421d-997a-b1a9787ae647">

**Comment by giovanninardini** [comment, marked as answer]
Please try to apply the same changes that you find in the latest commit - here: https://github.com/Unipisa/Simu5G/commit/a28aba469a785fcd166d2fd0c41408605d86d4b6 - or clone the updated master branch from the repo. 
Then build the code, run the simulation again and see if the problem arises again.

**Comment by rimsayegh** [comment]
Thank you for your guidance. I followed your instructions, and everything is working perfectly now.

Additionally, we are interested in running two applications, WarningAlert and VideoStreaming, simultaneously in the same scenario. Until now, we have been unable to execute both applications in parallel in the same scenario. Could you provide any insights or guidance on anyway to do so?

Thank you once again!

**Comment by giovanninardini** [comment]
Assuming you are using the SingleMecHost configuration, a UE runs a WarningAlertApp and a DeviceApp (which is used to contact the UALCMP module). If you want to add another application, you just need to add the new application and another DeviceApp.
See the following snippet of the INI configuration as example (running two WarningAlert apps):

```
*.ue[*].numApps = 4
*.ue[*].app[0].typename = "DeviceApp"
*.ue[*].app[0].localPort = 4500
*.ue[*].app[0].UALCMPAddress = "ualcmp"
*.ue[*].app[0].UALCMPPort = 1000
*.ue[*].app[0].appPackageSource = "ApplicationDescriptors/WarningAlertApp.json"

*.ue[*].app[1].typename = "UEWarningAlertApp"
*.ue[*].app[1].deviceAppAddress = "ue["+string(ancestorIndex(1))+"]"
*.ue[*].app[1].deviceAppPort = 4500
*.ue[*].app[1].localPort = 4000
*.ue[*].app[1].startTime = 1s
*.ue[*].app[1].stopTime = 30s	

*.ue[*].app[2].typename = "DeviceApp"
*.ue[*].app[2].localPort = 4501
*.ue[*].app[2].UALCMPAddress = "ualcmp"
*.ue[*].app[2].UALCMPPort = 1000
*.ue[*].app[2].appPackageSource = "ApplicationDescriptors/WarningAlertApp.json"

*.ue[*].app[3].typename = "UEWarningAlertApp"
*.ue[*].app[3].deviceAppAddress = "ue["+string(ancestorIndex(1))+"]"
*.ue[*].app[3].deviceAppPort = 4501
*.ue[*].app[3].localPort = 4001
*.ue[*].app[3].startTime = 2s 
*.ue[*].app[3].stopTime = 31s
```

Note the port numbers, which need to be different to prevent conflicts.
Moreover, if you use a different app, you also need to add the corresponding file in the ApplicationDescriptor folder.

**Reply by rimsayegh** [reply]
Thank you for your detailed explanation and guidance. I followed the example you provided, and it works perfectly.
